core/git.go

```diff
@@ -249,11 +236,14 @@ func (repo *RemoteGitRepository) setup(ctx context.Context) (_ *gitutil.GitCLI,
 		if err != nil {
 			return nil, nil, err
 		}
-		opts = append(opts, gitutil.WithArgs(
-			"-c", "http."+repo.URL.Remote()+".extraheader=Authorization: "+string(authHeader),
-		))
+		hl := strings.TrimSpace(string(authHeader))
+		if !strings.HasPrefix(strings.ToLower(hl), "authorization:") {
```

Comment:
🤔 why?

It's definitely an edge case, but there should be one behavior - we shouldn't attempt to read this as just a header line.